Cinema system/Cinema system/Controllers/CinemasController.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc;
+using CinemaSystem.Models;
+using Microsoft.EntityFrameworkCore;
+
+namespace CinemaSystem.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class CinemasController : ControllerBase
+    {
+
+        private CinemaDBContext _db;
+
+        public CinemasController(CinemaDBContext context)
+        {
+            _db = context;
+        }
+
+        [HttpGet]
+        public IEnumerable<Cinema> Get()
+        {
+            IEnumerable<Cinema> cinemas = _db.Cinemas.AsEnumerable();   
+            return cinemas;
+        }
+
+        [HttpGet("{id}")]
+        public ActionResult<Cinema> Get(int id)
+        {
+            var cinema = _db.Cinemas.Find(id);
+
+            if (cinema == null)
+            {
+                return NotFound();
+            }
+
+            return cinema;
+        }
+
+        [HttpPost]
+        public ActionResult<Cinema> Post(Cinema cinema)
+        {
+            if ((cinema == null) || !ModelState.IsValid)
+            {
+                return BadRequest();
```

Comment:
Shouldn't the client know what exactly he did wrong?

Reply:
The client will see the text of the error message that is described in the cinema class.